dotcom-rendering/src/web/browser/newsletterEmbedIframe/init.ts

```diff
@@ -1,33 +1,56 @@
 import '../webpackPublicPath';
 import { startup } from '@root/src/web/browser/startup';
 
-type NewsletterHeightEventType = { source: { location: { href: string } } };
+type MessageType = {
+	type: string;
+	value: string | number;
+};
 
+// No trailing slash!
+const allowedOrigins = ['https://www.theguardian.com'];
 const init = (): Promise<void> => {
 	const allIframes: HTMLIFrameElement[] = [].slice.call(
 		document.querySelectorAll('.email-sub__iframe'),
 	);
 
+	// Tell the iframes to resize once this script is loaded
+	// Otherwise, earlier resize events might be missed
+	// So we don't have to load this script as a priority on each load
+	allIframes.forEach((iframe) => {
+		if (iframe && iframe.contentWindow)
+			iframe.contentWindow.postMessage(
+				'resize',
+				'https://www.theguardian.com',
+			);
+	});
+
 	window.addEventListener('message', (event) => {
+		if (!allowedOrigins.includes(event.origin)) return;
+
 		const iframes: HTMLIFrameElement[] = allIframes.filter((i) => {
 			try {
-				return (
-					i.src ===
-					(event as NewsletterHeightEventType).source.location.href
-				);
+				if (i.contentWindow && event.source) {
+					return i.contentWindow === event.source;
+				}
 			} catch (e) {
 				return false;
 			}
+			return false;
 		});
 		if (iframes.length !== 0) {
 			try {
-				const message: { [key: string]: string } = JSON.parse(
-					event.data,
-				);
+				const message: MessageType = JSON.parse(event.data);
```

Comment:
Is there any reason not to perform the same `isObject` and `typeof` message checks here? I think your decision to annotate the result of `JSON.parse` as `unknown` in the AR version was very sensible.

Whenever you annotate a value that was previously of type `any`, that annotation is equivalent to a type cast (`as`). For example:

```ts
const a = JSON.parse('a'); // This is of type `any`
const b: number = a; // This compiles fine
```

In the above example, `b` can be passed anywhere a `number` is expected, which will likely cause runtime errors. One way to solve this problem, as seen in your AR version, is to annotate with `unknown`. That's still a type cast, but it's saying that we known absolutely nothing about that value, so it can't really be used anywhere without causing compiler errors. In order to use it TS will force us to make checks, like `typeof` and `isObject`, to figure out the correct type.

**Note:** If you choose to make this change, in DCR I think `isObject` is available from `@guardian/libs`. The reason it's in the local `lib` in AR is because I believe its existence there predates the introduction to `@guardian/libs`.

George, I wouldn't worry about this for your PR, but I'm concerned that these two implementations (DCR and AR) have already diverged. I believe the two modules are meant to be operating on the same embeds and providing the same functionality? @oliverlloyd I'm aware it was your suggestion to keep them separate and not share the implementation, but to my mind capturing these kind of rules in one place and keeping them in sync is important 🤔

That said, I'm not suggesting you have to refactor things again for now George! I think it's something for the team to consider more broadly.

Reply:
> I'm aware it was your suggestion to keep them separate and not share the implementation, but to my mind capturing these kind of rules in one place and keeping them in sync is important

Sorry, I didn't mean to say I'm against sharing this code. In principle, that makes sense, esp. as you say, we're dealing with the same embeds here!

However, I don't think `common-rendering` is the right place to do it. It's not setup for sharing lib code and once you start to do something more will follow - lib folders have a tendency to grow messy fast!

Rather than create a quick abstraction here, I'd rather we take an approach where we stick with duplication for now and then abstract slower using @guardian/libs 

A slower abstraction allows time to consider what is needed and what, if any, shared areas there are that could be abstracted and what the best api for that abstraction might be. Other benefits to guardian/libs is it is a place that allows wider contributions and tends to have a higher bar to entry, which, in my opinion, is a very good thing when creating abstractions

Reply:
> Is there any reason not to perform the same isObject and typeof message checks here?

No, it was mostly resistance on my part because of `isObject` living in AR. I had not realised it is in `@guardian/libs`, so I've implemented the same checks in DCR as well now.